astropy/modeling/core.py

```diff
@@ -940,6 +1073,106 @@ def bounding_box(self, limits):
     def evaluate(self, *args, **kwargs):
         """Evaluate the model on some input variables."""
 
+    def render(self, out=None, coords=None):
+        """
+        Evaluates a model on an input array. Evaluation is limited to
+        a bounding box if the `Model.bounding_box` attribute is set.
+
+        Parameters
+        ----------
+        out : `numpy.ndarray`, optional
+            The array on which the model is to be evaluated.
+        coords : array-like, optional
+            Coordinate arrays mapping to ``arr``, such that
+            ``arr[coords] == arr``.
+
+        Returns
+        -------
+        out : `numpy.ndarray`
+            The model evaluated on the input array if given, or else a new array from
+            ``coords``.
+            If ``out`` and ``coords`` are both `None`, the returned array is
+            limited to the `Model.bounding_box` limits. If
+            `Model.bounding_box` is `None`, ``arr`` or ``coords`` must be passed.
+
+        Examples
+        --------
+        :ref:`bounding-boxes`
+        """
+
+        try:
+            bbox = self.bounding_box
+        except NotImplementedError:
+            bbox = None
+
+        ndim = self.n_inputs
+
+        if (coords is None) and (out is None) and (bbox is None):
+            raise ValueError('If no bounding_box is set, '
+                             'coords or out must be input.')
+
+        # for consistent indexing
+        if ndim == 1:
+            if coords is not None:
+                coords = [coords]
+            if bbox is not None:
+                bbox = [bbox]
+
+        if coords is not None:
+            # Check dimensions match out and model
+            assert len(coords) == ndim
+            if out is not None:
+                assert coords[0].shape == out.shape
+            else:
+                out = np.zeros(coords[0].shape)
+
+        if out is not None:
+            try:
+                assert out.ndim == ndim
+            except AssertionError:
+                raise AssertionError(
+                    'The array and model must have the same number '
+                    'of dimensions.')
+
```

Comment:
At some point something needs to be done with these assertions--really there's no reason to use assert and then catch AssertionErrors and reraise them.  This should just raise a ValueError something after checking the condtion, and not use `assert` at all.  I don't think I have time to fix it now though.